src/domain/service/noteRelationship.ts

```diff
@@ -0,0 +1,34 @@
+import type { Note } from '@domain/entities/note.js';
+import type NoteRelationshipRepository from '@repository/noteRelationship.repository.js';
+
+/**
+ * Note relationship service
+ */
+export default class NoteRelationshipService {
+  /**
+   * Note relationship repository
+   */
+  public repository: NoteRelationshipRepository;
+
+  /**
+   * Note relationship service constructor
+   *
+   * @param repository - note relationship repository
+   */
+  constructor(repository: NoteRelationshipRepository) {
+    this.repository = repository;
+  }
+
+  /**
+   * Adds note relationship
+   *
+   * @param noteId - id of the current note
+   * @param parentId - id of the parent note
+   */
+  public async addNoteRelation(noteId: Note['id'], parentId: Note['id']): Promise<void> {
```

Comment:
I think, that you should return something here to ensure that the record has been created

Reply:
> I think, that you should return something here to ensure that the record has been created

Ok, how about `boolean` variable?

Reply:
it will be ok